src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -3803,22 +3803,47 @@ mono_profiler_jit_done (
 	MonoMethod *method,
 	MonoJitInfo *ji)
 {
-	if (!EventEnabledMonoProfilerJitDone())
+	if (!EventEnabledMonoProfilerJitDone() && !EventPipeEventEnabledMonoProfilerJitDoneVerbose())
 		return;
 
+	bool verbose = (MICROSOFT_DOTNETRUNTIME_MONO_PROFILER_PROVIDER_EVENTPIPE_Context.Level >= (uint8_t)EP_EVENT_LEVEL_VERBOSE);
+
 	uint64_t method_id;
 	uint64_t module_id;
 	uint32_t method_token;
 
 	get_jit_data (method, &method_id, &module_id, &method_token);
 
-	FireEtwMonoProfilerJitDone (
-		clr_instance_get_id (),
-		method_id,
-		module_id,
-		method_token,
-		NULL,
-		NULL);
+	if (verbose) {
```

Comment:
This is a little bit odd.  When you turn on a provider with a specific keyword and level, the expectation is that all events associated with that keyword and level (or below) are enabled.  Is it possible to turn both of these events on by keyword/level?  If so, then it might be worth setting these up in different keywords so that users can turn one or the other on.  The idea that if one event is on then the other is off doesn't really match the control mechanism, but I get your point that its not necessary to have both on at the same time.  I think the key here is that you don't want someone switching to verbose because they need it for some other keyword, and losing an event that they expected to get based on the configuration.

Reply:
This is how it works for the method load events like FireEtwMethodLoadVerbose_V1, FireEtwMethodLoadVerbose_V2, if you setup to verbose you will get different event compared to not running with normal log level, https://github.com/dotnet/runtime/blob/921bca0ae0d07a7dfd83d5d70eaf49759ab8463b/src/coreclr/vm/eventtrace.cpp#L6815

Reply:
I guess I wouldn't introduce that as a new pattern, but since you're following the pattern already here, I guess that's fine. :)

Reply:
Have give this a second though, will look at including this event (and use it for another use case in Mono as well) under the tracing keyword, and probably rename it a little, stay tuned.